File: 485.py
"""
1. use a stack to store the 1s
or,
2. use a count variable to store the length only
"""
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxLength = 0
        storeList = []
        nums += [0]
        for i in nums:
            if not i:
                maxLength = max(maxLength, len(storeList))
                storeList = []
            else:
                storeList.append(i)
        return maxLength


class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxLength = 0
        length = 0
        nums += [0]
        for i in nums:
            if not i:
                maxLength = max(maxLength, length)
                length = 0
            else:
                length += 1
        return maxLength

File: solution/go/485.go
/*
If 0, reset cur, O(n) time
*/
func findMaxConsecutiveOnes(nums []int) int {
    max, cur := 0, 0
    for _, val := range nums {
        if val == 0 {
            cur = 0
        } else {
            cur += 1
            if cur > max {
                max = cur
            }
        }
    }
    return max
}
